phoenix/core/models.py

```diff
@@ -185,33 +147,48 @@ def is_resolved(self):
     def is_reopened(self):
         return self.solution is not None and not self.resolved
 
-    @property
-    def resolved_on_time(self):
-        now = arrow.utcnow()
-        solution = self.is_resolved
-        if solution:
-            if solution.created > self.eta_human_deadline:
-                return False
-            return True
-
-        if self.eta_human_deadline < now:
-            return False
-        return True
-
     def set_eta(self, eta):
         """Properly sets eta. Never change eta manually.
 
         This method will set eta and eta_last_modifie values.
         Both are needed for proper representation.
         """
-        if not eta:
-            # check because slack can return null
-            eta = 0
-        self.eta_last_modified = timezone.now()
         self.eta = eta
-        if eta:
-            self.prompt_for_eta_update = False
-            self.prompt_active = False
+        self.eta_last_modified = timezone.now()
+
+    @property
+    def eta_in_minutes(self):
+        granularity_table = {"m": 1, "h": 60}
+        m = ETA_PARSE_RE.match(self.eta)
+        value = m.groupdict()["value"]
+        granularity = m.groupdict()["granularity"]
+        symbol = m.groupdict()["symbol"]
+        if symbol == ">":
+            # symbol ">" is only used for last value that is represented as unknown ETA
+            return None
+        x = granularity_table[granularity]
+        return int(value) * x
+
+    @property
+    def eta_deadline(self):
+        eta_in_minutes = self.eta_in_minutes
+        if not eta_in_minutes:
+            return ""
+        eta_last_modified = arrow.get(self.eta_last_modified)
+        return eta_last_modified.shift(minutes=eta_in_minutes)
+
+    @property
+    def eta_remaining(self):
+        """Calculate remaining ETA from now in minutes."""
+        eta_deadline = self.eta_deadline
+        if not eta_deadline:
+            return ""
+        delta = eta_deadline - arrow.utcnow()
+        minutes, _ = divmod(delta.total_seconds(), 60)
```

Comment:
This one is equivalent to `minutes = delta.total_seconds() // 60` which is simpler

Reply:
Changed to `//`.